engine/evd.ml

```diff
@@ -1393,7 +1399,7 @@ let new_pure_evar ?(src=default_source) ?(filter = Filter.identity) ?(relevance
   }
   in
   let typeclass_candidate = if principal then Some false else typeclass_candidate in
-  let newevk = new_untyped_evar () in
+  let newevk = Evar.unsafe_of_int evd.evar_counter in
```

Comment:
I liked the API to allocate a new evar.
In the other patch, you do incr it. In this branch, I don't even see where you do it.